nomos-services/blend/src/core/state.rs

```diff
@@ -163,6 +227,62 @@ mod service {
             self.spent_core_quota
         }
 
+        pub(super) fn update_token_collector(
+            &mut self,
+            collector: SessionBlendingTokenCollector,
+        ) -> Result<(), SessionMismatchError> {
+            Self::validate_token_collector(self.last_seen_session, &collector)?;
+            self.token_collector = collector;
+            Ok(())
+        }
+
+        const fn validate_token_collector(
+            last_seen_session: SessionNumber,
+            collector: &SessionBlendingTokenCollector,
+        ) -> Result<(), SessionMismatchError> {
+            if last_seen_session == collector.session_number() {
+                Ok(())
+            } else {
+                Err(SessionMismatchError {
+                    last_seen: last_seen_session,
+                    actual: collector.session_number(),
+                })
+            }
+        }
+
+        pub(super) fn update_old_session_token_collector(
+            &mut self,
+            collector: OldSessionBlendingTokenCollector,
+        ) -> Result<(), SessionMismatchError> {
+            Self::validate_old_session_token_collector(self.last_seen_session, &collector)?;
+            self.old_session_token_collector = Some(collector);
+            Ok(())
+        }
+
+        const fn validate_old_session_token_collector(
```

Comment:
Do we need to separate functions, or can we simply use a smaller session number for the previous session collector?

Reply:
Fair enough, thanks for the explanation.